src/scales/scale.time.js

```diff
@@ -625,21 +625,46 @@ module.exports = function(Chart) {
 			return ticksFromTimestamps(ticks, me._majorUnit);
 		},
 
-		getLabelForIndex: function(index, datasetIndex) {
+		getLabelForIndex: function(index) {
 			var me = this;
 			var data = me.chart.data;
 			var timeOpts = me.options.time;
 			var label = data.labels && index < data.labels.length ? data.labels[index] : '';
-			var value = data.datasets[datasetIndex].data[index];
+			var len = data.labels.length;
```

Comment:
I think it should use `me._timestamps.data` instead to also take in account labels embedded in object data (`{t, y}`).